BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -218,56 +271,56 @@ public void onRequestPermissionsResult(int requestCode,
     }
   }
 
-  private void createBibFile(String folderName, String fileName) {
+  private void bibExportShelf(String fileName) {
+    String rootPathStr = Environment.getExternalStorageDirectory() + File.separator
+        + folderName + File.separator + fileName + fileTypeBib;
     try {
-      String rootPath = Environment.getExternalStorageDirectory() + "/" + folderName + "/";
-      File root = new File(rootPath);
-
-      if (!root.exists()) {
-        root.mkdirs();
-      }
-
-      File file = new File(rootPath + fileName + ".bib");
+      File file = new File(rootPathStr);
       if (file.exists()) {
         file.delete();
       }
-      file.createNewFile();
 
+      file.createNewFile();
       FileOutputStream out = new FileOutputStream(file);
       out.flush();
       out.close();
 
-      Toast.makeText(getContext(), getString(R.string.exported_file_stored_in) + '\n'
-          + "/" + folderName + "/" + fileName + ".bib", Toast.LENGTH_LONG).show();
+      writeBibFile(shelfName);
 
     } catch (Exception e) {
       e.printStackTrace();
     }
+
+    Toast.makeText(getContext(),
+        getString(R.string.exported_file_stored_in) + '\n'
+            + File.separator + folderName + File.separator + fileName
+            + fileTypeBib, Toast.LENGTH_LONG).show();
   }
 
-  private void retrieveBibContent() {
+  private String retrieveBibContent() {
 
-    DatabaseHelper dbHelper = new DatabaseHelper(getContext());
-    BookDao bd = new BookDao(dbHelper);
-    NoteDao nd = new NoteDao(dbHelper);
+    dbHelper = new DatabaseHelper(getContext());
+    bookDao = new BookDao(dbHelper);
+    NoteDao noteDao = new NoteDao(dbHelper);
 
-    List<Long> bookIdsCurrShelf = bd.getAllBookIdsForShelf(shelfId);
+    List<Long> bookIdsCurrShelf = bookDao.getAllBookIdsForShelf(shelfId);
     List<Author> authorsCurrBook = new ArrayList<>();
     List<Long> notesCurrBook = new ArrayList<>();
     String allNotesCurrBook;
     String authorNamesCurrBook;
+    String bibShelfContent = "";
 
     //for each book in the current shelf
```

Comment:
Redundant comment

Reply:
do you mean bibShelfContent  (line 311) or the comment in line 313?

Reply:
Sorry and yes the comment in line 313.

Reply:
I would leave it, just because it helps me to understand my code later, when I will more data to the bibTex.